### PR TITLE
Full test with install combined -> nightwatch

### PR DESCRIPTION
**Pull request by dawehner**
Based upon https://github.com/jsdrupal/drupal/pull/3

### CODE REVIEW

File: core/tests/Drupal/Nightwatch/Commands/installDrupal.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+const exec = require('child_process').exec;
+
+/**
+ * @param browser
+ * @param cookieValue
+ * @returns {*}
+ */
+const setupCookie = function (browser, cookieValue, done) {
+  const matches = process.env.BASE_URL.match(/^https?\:\/\/([^\/?#]+)(?:[\/?#]|$)/i);
```
Comment (mattgrill) [inline core/tests/Drupal/Nightwatch/Commands/installDrupal.js:9]: Couldn't we use the build in `url` module?

File: core/tests/Drupal/Nightwatch/Commands/installDrupal.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      domain: domain,
+    }, done);
+};
+
+exports.command = function installDrupal(profile = 'testing', setupFile = '', done) {
+  const self = this;
+
+  exec(`php ./scripts/setup-drupal-test.php ${profile} --setup_file ${setupFile}`, (err, simpleTestCookie) => {
+    if (err) {
+      console.error(err);
```
Comment (mattgrill) [inline core/tests/Drupal/Nightwatch/Commands/installDrupal.js:30]: Do we want to throw an error here if the install command fails?

File: core/tests/Drupal/Nightwatch/Tests/index.js [hunk 3]
```diff
@@ -1,9 +1,10 @@
 module.exports = {
-  'Demo Drupal.org': (browser) => {
+  'Test page': (browser, done) => {
```
Comment (mattgrill) [inline core/tests/Drupal/Nightwatch/Tests/index.js:2]: What is `done`?

File: core/tests/Drupal/Nightwatch/Tests/index.js [hunk 4]
```diff
@@ -1,9 +1,10 @@
...
     browser
-      .relativeURL('/community')
+      .installDrupal('testing', __dirname + '/index.setup.php')
```
Comment (mattgrill) [inline core/tests/Drupal/Nightwatch/Tests/index.js:4]: Is there any way we can make a guess about what this setup file will be? I think if we can alleviate potential pain points for test writers that will be wonderful.

File: core/tests/Drupal/Nightwatch/globals.js [hunk 5]
```diff
@@ -1,7 +1,16 @@
 const chromedriver = require('chromedriver');
+const settings = require('../../../nightwatch.settings.json');
 
 module.exports = {
   before: (done) => {
+    // Setting up 
+    const baseUrl = settings.BASE_URL || process.env.SIMPLETEST_BASE_URL || process.env.BASE_URL;
```
Comment (mattgrill) [inline core/tests/Drupal/Nightwatch/globals.js:7]: Do we need to check to see that these environment variables are possibly empty?

File: core/tests/Drupal/Nightwatch/Tests/index.js (hunk 3 above)
Comment (dawehner) [reply core/tests/Drupal/Nightwatch/Tests/index.js:2]: its like cb :)

File: core/tests/Drupal/Nightwatch/Tests/index.js (hunk 4 above)
Comment (dawehner) [reply core/tests/Drupal/Nightwatch/Tests/index.js:4]: I agree, but well, the function call happens in a different site, so I'm not sure

File: core/tests/Drupal/Nightwatch/globals.js (hunk 5 above)
Comment (dawehner) [reply core/tests/Drupal/Nightwatch/globals.js:7]: I'm curious, does || prevent against that?

**Pull request merged**